lua/starfall/permissions/providers_sh/url_whitelist.lua

```diff
@@ -2,14 +2,19 @@
 
 local urlrestrictor = SF.StringRestrictor(false)
 
-local function checkWhitelist(instance, target, key)
-	if TypeID(target) ~= TYPE_STRING then return false, "The url is not a string" end
-	local prefix = string.match(target,"^(%w-)://") -- Check if protocol was given
-	if not prefix then -- If not, add http://
-		target = "http://"..target
+local function checkWhitelist(instance, url, key)
+	if TypeID(url) ~= TYPE_STRING then return false, "The url is not a string" end
+
+	if not string.match(url,"^(%w-)://") then
+		url = "http://"..url
+	end
+
+	local result = gamemode.Call("CanAccessUrl", url)
```

Comment:
I think this will need to be `hook.Run`, `gamemode.Call` skips hooks not defined on the gamemode table.

Reply:
Fixed